cargo-dylint/tests/ci.rs

```diff
@@ -133,6 +133,169 @@ fn cargo_dylint_and_dylint_readmes_are_equal() {
     compare_lines(&cargo_dylint_readme, &dylint_readme);
 }
 
+#[test]
+fn examples_readme_contents() {
+    let examples_dir = Path::new(env!("CARGO_MANIFEST_DIR")).join("../examples");
+    let categories = vec![
+        "general",
+        "supplementary",
+        "restriction",
+        "experimental",
+        "testing",
+    ];
+
+    // Read the existing README
+    let readme_path = examples_dir.join("README.md");
+    let readme_content = read_to_string(&readme_path).unwrap();
+
+    // Generate just the lint description tables
+    let expected_tables = generate_lint_tables(&examples_dir, &categories);
+
+    // Extract the current tables section from README using markers
+    let actual_tables = extract_between_markers(&readme_content)
+        .unwrap_or_else(|| panic!("Lint description markers not found in README.md"));
+
+    // Compare the trimmed generated content with the actual content
+    assert_eq!(
+        expected_tables.trim(),
+        actual_tables,
+        "Lint descriptions in README.md do not match expected content"
+    );
+}
+
+fn generate_lint_tables(examples_dir: &Path, categories: &[&str]) -> String {
+    let mut content = String::new();
+
+    // Generate the tables for each category
+    for category in categories {
+        use std::cmp::max;
+        use std::fmt::Write;
+
+        // Get the examples for this category
+        let examples = collect_examples_from_category(examples_dir, category);
+
+        // Calculate column widths
+        let example_header = "Example";
+        let desc_header = "Description/check";
```

Comment:
```suggestion
        const EXAMPLE_HEADER: &str = "Example";
        const DESC_HEADER: &str = "Description/check";
```
Nit. And these will probably have to be moved to earlier in the function body.